### PR TITLE
[CI] Fix link checker

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: .github/workflows/valid-links.yml [hunk 1]
```diff
@@ -14,6 +14,7 @@ jobs:
       - run: |
           git fetch --prune origin master
           git checkout $GITHUB_SHA
+          git rebase origin/master
```
Comment (ErikSchierboom) [inline .github/workflows/valid-links.yml]: This step is required to have `git diff` only output the actual changes made in this branch. Indidentally, this was probably the reason why the link checker failed in #234, as the diff could show files that were deleted.

File: .github/workflows/valid-links.yml [hunk 2]
```diff
@@ -22,13 +23,13 @@ jobs:
         run: npm install -g markdown-link-check
       - name: Verify links
         run: |
-          deleted_or_renamed=$(git diff --no-commit-id --name-only --diff-filter DR origin/master | grep  -i .md$ | grep -v -i _sidebar.md | grep -v -i ISSUE_TEMPLATE | wc -l || true)
+          deleted_or_renamed=$(git diff --no-commit-id --name-only --diff-filter DR origin/master | grep  -i .md$ | grep -v -i _sidebar.md | grep -v -i ISSUE_TEMPLATE | wc -l)
```
Comment (ErikSchierboom) [inline .github/workflows/valid-links.yml]: The command ending with `wc -l` makes sure the exit code is always 0.

File: .github/workflows/valid-links.yml [hunk 3]
```diff
@@ -22,13 +23,13 @@ jobs:
...
 
           if [ $deleted_or_renamed -eq 0 ]
           then
-              files=$(git diff --no-commit-id --name-only --diff-filter AM origin/master | grep  -i .md$ | grep -v -i _sidebar.md | grep -v -i ISSUE_TEMPLATE || true)
+              files=$(git diff --no-commit-id --name-only --diff-filter AM origin/master | grep  -i .md$ | grep -v -i _sidebar.md | grep -v -i ISSUE_TEMPLATE | cat)
```
Comment (ErikSchierboom) [inline .github/workflows/valid-links.yml]: This is a far more elegant solution to fixing the exit code when the grep can't find any results.

File: .github/workflows/valid-links.yml [hunk 4]
```diff
@@ -14,6 +14,9 @@ jobs:
       - run: |
           git fetch --prune origin master
           git checkout $GITHUB_SHA
+          git config --global user.email "ci@exercism.io"
```
Comment (ErikSchierboom) [inline .github/workflows/valid-links.yml]: This is required for the rebase

File: .github/workflows/valid-links.yml [hunk 5]
```diff
@@ -22,13 +25,13 @@ jobs:
         run: npm install -g markdown-link-check
       - name: Verify links
         run: |
-          deleted_or_renamed=$(git diff --no-commit-id --name-only --diff-filter DR origin/master | grep  -i .md$ | grep -v -i _sidebar.md | grep -v -i ISSUE_TEMPLATE | wc -l || true)
+          deleted_or_renamed=$(git diff --no-commit-id --name-only --diff-filter DR origin/master | grep  -i .md$ | grep -v -i _sidebar.md | grep -v -i ISSUE_TEMPLATE | wc -l)
 
-          if [ $deleted_or_renamed -eq 0 ]
+          if [ $GITHUB_REF = "master" ] || [ $deleted_or_renamed -ne 0 ]
```
Comment (ErikSchierboom) [inline .github/workflows/valid-links.yml]: This should ensure the CI is running on scheduled workflows

File: .github/workflows/valid-links.yml [hunk 6]
```diff
@@ -4,16 +4,20 @@ on:
...
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
       - run: |
-          git fetch --prune origin master
-          git checkout $GITHUB_SHA
+          git config --global user.email "action@github.com"
+          git config --global user.name "GitHub Action"
```
Comment (SaschaMann) [inline .github/workflows/valid-links.yml]:
```suggestion
          git config --global user.name "github-actions[bot]"
```

To keep it consistent with the bot that is used when you use the Actions Token.

File: .github/workflows/valid-links.yml [hunk 7]
```diff
@@ -4,16 +4,20 @@ on:
...
   verify-links:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
       - run: |
-          git fetch --prune origin master
-          git checkout $GITHUB_SHA
+          git config --global user.email "action@github.com"
```
Comment (SaschaMann) [inline .github/workflows/valid-links.yml]:
```suggestion
          git config --global user.email "github-actions[bot]@users.noreply.github.com"
```

**Comment by tehsphinx** [comment]
@SaschaMann beat me to it: just wanted to say that I won't be of much help here without some large time investment... and then I'd do a lot of trial and error...

**Comment by ErikSchierboom** [comment]
@iHiD Could you look at this?

**Pull request merged**